resources/faq.md

```diff
@@ -0,0 +1,163 @@
+
+MeiliSearch is an **asynchronous API**.<br>
+It means that, in a lot of cases (e.g., documents addition), you will receive as server response a simple JSON with only an `updateId` attribute. For example:
+
+```json
+{
+    "updateId": 2
+}
+```
+
+This kind of **successful response** indicates that the operation has been taken into account, but it may not have been executed yet.<br>
+
+::: tip
+You can check the status of the operation via this `updateId` and the [get update status route](/references/updates.md#get-an-update-status).<br>
+Also, MeiliSearch provides a global route to [track all your update status](/references/updates.md#get-all-update-status).<br>
+**Thus, you will know if your action succeeded or not, and <u>why</u>**.
+:::
+
+If you are curious about how the asynchronous part of MeiliSearch works, here is [more information](/guides/advanced_guides/asynchronous_updates.md).
+
+## I try to add my documents but I still get a `400 - Invalid data` response.
+
+MeiliSearch API only accepts JSON format.<br>
+And in the case of the [document addition route](/references/documents.md#add-or-replace-documents), only an array of objects is expected.
+
+The `400 - Invalid data` response probably means that your data is not in an expected format.
+
+Most common errors:
+- you let a useless coma at the end of a line.
+- your data is not an array of objects: for the [document addition route](/references/documents.md#add-or-replace-documents), MeiliSearch only accepts an array in the body even if there is only one document.
+
+Wrong:
+```json
+{ "id": 123,  "title": "Pride and Prejudice" }
+```
+
+Good:
+```json
+[
+    { "id": 123,  "title": "Pride and Prejudice" }
+]
+```
+
+:::tip
+The [jq](https://github.com/stedolan/jq) command line tool can greatly help you check the format of your data.
+```bash
+$ cat your_file.json | jq
+```
+:::
+
+## My document uploading failed with the `document id is missing` error.
+
+::: note TLDR;
+Most common reasons:
+- You have not put any unique identifier in your document.
+- The unique identifier of your document is not well-formatted.
+:::
+
+MeiliSearch needs each document to contain a unique identifier. This identifier attribute is the `primary key`.
+
+How do I know the primary key of my index? [Check this route](/references/indexes.md#get-one-index). The `null` value means it has not been defined yet.
+
+By default, MeiliSearch will infer the primary key in the first document sent. It will search for an attribute that contains the string `id` in any way case-insensitively (e.g., `uid`, `MovieId`, `ID`, `123id123`). If none has been found, no documents will be added.
+
+If you get `document id is missing` error, it means MeiliSearch could not recognize your primary key. This means your primary key does not answer the formatting explained above. To solve this error, you could send the primarykey's name as a query parameter [when adding documents](/references/documents.md#add-or-replace-documents).
```

Comment:
```suggestion
If you get a `document id is missing` error, the primary key was not recognized. This means your primary key is wrongly formatted. Sending primarykey's name as a query parameter [when adding documents](/references/documents.md#add-or-replace-documents) should solve this issue.
```

Reply:
I've just realize that I did a mistake: `primarykey's` instead of `primary key's`. Can you do a new suggestion with this fix? :)

Reply:
Didn't see sorry, it is done!